notebooks/ndd_example.ipynb

```diff
@@ -0,0 +1,202 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "bbf93d45-bee3-4202-8d5a-88497196db06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import requests\n",
+    "import warnings\n",
+    "import urllib3\n",
+    "warnings.filterwarnings(\"ignore\", category=urllib3.exceptions.InsecureRequestWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "06e9a476-134f-4901-a89e-3c91bdb8881c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_model_id():\n",
+    "    query = {\n",
+    "        'query': {\n",
+    "            'bool': {\n",
+    "                'must': [\n",
+    "                    {\n",
+    "                        'match': {'name': 'all-MiniLM-L6-v2'},\n",
+    "                    },\n",
+    "                    {\n",
+    "                        'term': {'model_config.model_type': 'bert'},\n",
+    "                    },\n",
+    "                ],\n",
+    "            },\n",
+    "        },\n",
+    "    }\n",
+    "    with requests.get('https://opensearch:9200/_plugins/_ml/models/_search', json=query, verify=False) as resp:\n",
+    "        res = json.loads(resp.text)\n",
+    "        return res['hits']['hits'][0]['_id']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "4a68af4e-843e-4c4e-b295-1a4e9ad55881",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def do_query(query_dict):\n",
+    "    url = 'https://opensearch:9200/demoindex0/_search?search_pipeline=hybrid_rag_pipeline'\n",
+    "    with requests.post(url, json=query, verify=False) as resp:\n",
+    "        res = json.loads(resp.text)\n",
+    "        hits = res['hits']['hits']\n",
+    "        for i in range(10):\n",
+    "            text = hits[i]['_source']['text_representation']\n",
+    "            text = text.replace('\\n', ' ')[:80]\n",
+    "            print(f'[{i}] {text}')\n",
+    "        answer = res['ext']['retrieval_augmented_generation']['answer']\n",
+    "        print(f'[ANSWER]\\n{answer}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61a71039-0a85-4e14-b42a-0969d04d979a",
+   "metadata": {},
+   "source": [
+    "First, we run the query without near-duplicate-detection.  We do this by not asking for `shingles` in `_source`.  The results we get have many copies of the first chunk.  The generated answer comes from only one chunk."
```

Comment:
I don't think we use this syntax anywhere in examples or our docs for OpenSearch queries. You might need to explain what 'source' is in the query.